File: src/main/java/me/jsinco/solutilities/utility/Saves.java
package me.jsinco.solutilities.utility;

import org.bukkit.Bukkit;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;

import java.io.File;
import java.io.IOException;

public class Saves {
    private static File file;
    private static FileConfiguration customFile;

    //Finds or generates YAML used for storing how many points each player has using UUIDs as keys
    public static void setup(){
        file = new File(Bukkit.getServer().getPluginManager().getPlugin("SolUtilities").getDataFolder(), "Saves.yml");

        if (!file.exists()){
            try{
                file.createNewFile();
            }catch (IOException e){
                //ow
            }
        }
        customFile = YamlConfiguration.loadConfiguration(file);
    }

    public static FileConfiguration get(){
        return customFile;
    }

    public static void save(){
        try{
            customFile.save(file);
        }catch (IOException e){
            //
        }
    }

    public static void reload(){
        customFile = YamlConfiguration.loadConfiguration(file);
    }
}

File: src/main/java/me/jsinco/solutilities/Saves.kt
package me.jsinco.solutilities

import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

object Saves {
    private val filemanager = FileManager("data/saves.yml")
    private val yamlSaves = filemanager.getFileYaml()

    @JvmStatic
    fun setup() {
        filemanager.generateFolder("data")
        filemanager.generateFile()
    }

    @JvmStatic
    fun get(): YamlConfiguration {
        return yamlSaves
    }

    @JvmStatic
    fun save() {
        filemanager.saveFileYaml()
    }

    @JvmStatic
    fun reload() {
        filemanager.reloadFileYaml()
    }
}